pyctuator/httptrace/http_tracer.py

```diff
@@ -11,4 +12,16 @@ def get_httptrace(self) -> Traces:
         return Traces(list(self.traces_list))
 
     def add_record(self, record: TraceRecord) -> None:
+
+        self._scrub_and_normalize_headers(record.request.headers)
+        self._scrub_and_normalize_headers(record.response.headers)
+
         self.traces_list.append(record)
+
+    def _scrub_and_normalize_headers(self, headers: dict) -> None:
+        if headers:
+            for k, values in headers.items():
+                if isinstance(values, list):
+                    headers[k] = [scrub_header_value(k, v) for v in values]
+                else:
+                    headers[k] = [scrub_header_value(k, values)]
```

Comment:
So mypy correctly complains on type mismatch here. Once complaint is that `headers` is `Mapping[str, List[str]]` and not `dict`. Another complaint is that the value is always a list.

Also, this function will be better if instead of modifying the input parameter, it would return a new mapping:
```
    def _scrub_and_normalize_headers(self, headers: Mapping[str, List[str]]) -> Mapping[str, List[str]]:
        if not headers:
            return headers

        return {header: [scrub_header_value(header, value) for value in values] for (header, values) in headers.items()}
```

Of course line 16,17 above need to change to:
```
        record.request.headers = self._scrub_and_normalize_headers(record.request.headers)
        record.response.headers = self._scrub_and_normalize_headers(record.response.headers)
```